test/util/testnode/read_test.go

```diff
@@ -0,0 +1,62 @@
+package testnode
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestReverseSlice(t *testing.T) {
```

Comment:
[optional] `TestReverseSlice` makes me think that this tests the exported function `ReverseSlice`. But it actually tests `reverseSlice` so this could be

```suggestion
func Test_reverseSlice(t *testing.T) {
```